src/ReactWrapper.js

```diff
@@ -525,6 +526,9 @@ export default class ReactWrapper {
    * @returns {Boolean}
    */
   hasClass(className) {
+    if (className instanceof RegExp) {
```

Comment:
`instanceof` isn't reliable across realms, and isn't actually a spec-compliant "is regex" test. https://www.npmjs.com/package/is-regex would fix the former.